workflows/hic/hicpro-core.wdl

```diff
@@ -0,0 +1,1103 @@
+        ndbup=$(( $allcount - $allcount_rmdup ))
+
+        ## merge stat file
+        echo -e "valid_interaction\t"$allcount > ~{prefix}_allValidPairs.mergestat
+        echo -e "valid_interaction_rmdup\t"$allcount_rmdup >> ~{prefix}_allValidPairs.mergestat
+        awk 'BEGIN{cis=0;trans=0;sr=0;lr=0} $2 == $5{cis=cis+1; d=$6>$3?$6-$3:$3-$6; if (d<=20000){sr=sr+1}else{lr=lr+1}} $2!=$5{trans=trans+1}END{print "trans_interaction\t"trans"\ncis_interaction\t"cis"\ncis_shortRange\t"sr"\ncis_longRange\t"lr}' ~{prefix}.allValidPairs >> ~{prefix}_allValidPairs.mergestat
+
+        if ~{if defined(capture_target) then "true" else "false"}
+        then
+            python /HiC-Pro_3.0.0/scripts/onTarget.py \
+                -i ~{prefix}.allValidPairs \
+                -t ~{capture_target} \
+                ~{if (report_capture) then "--cis" else ""} \
+                -s ~{prefix}_allValidPairs.mergestat \
+                -v \
+                > ~{prefix}_ontarget.allValidPairs
+        fi
+
+        if ~{allele_specific_snp}
+        then
+            python /HiC-Pro_3.0.0/scripts/split_valid_interactions.py \
+                ~{(
+                    if defined(capture_target)
+                    then "-i  ~{prefix}_ontarget.allValidPairs"
+                    else "-i ~{prefix}.allValidPairs"
+                )} \
+                -s ~{prefix}_allValidPairs_assplit.stat \
+                -v    
+        fi
+    >>>
+
+    output {
+        File all_valid_pairs = prefix + ".allValidPairs"
+        File? on_target = prefix + "_ontarget.allValidPairs"
+        File? split_interactions = prefix + "_allValidPairs_assplit.stat"
+        File all_valid_pairs_stats = prefix + "_allValidPairs.mergestat"
+    }
+
+    runtime {
+        container: "nservant/hicpro:3.0.0"
+        maxRetries: 1
+    }
+}
+
+task merge_stats {
+    meta {
+        description: "Merge statistics files"
+        outputs: {
+            read1_mapping_stats_merged: "Mapping statistics for read 1",
+            read2_mapping_stats_merged: "Mapping statistics for read 2",
+            valid_pairs_stats_merged: "Valid pairs statistics",
+            rs_stats_merged: "RS statistics",
+        }
+    }
+
+    parameter_meta {
+        read1_mapping_stats: "Mapping statistics for read 1"
+        read2_mapping_stats: "Mapping statistics for read 2"
+        valid_pairs_stats: "Valid pairs statistics"
+        rs_stats: "RS statistics"
+        prefix: "Prefix for the output files"
+    }
+
+    input {
+        Array[File] read1_mapping_stats
+        Array[File] read2_mapping_stats
+        Array[File] valid_pairs_stats
+        Array[File] rs_stats
+        String prefix
+    }
+
+    command <<<
+        set -euo pipefail
+
+        # merge read1 mapping stats
+        mkdir read1
+        ln -s ~{sep(" ", read1_mapping_stats)} read1/
+        python /HiC-Pro_3.0.0/scripts/merge_statfiles.py \
+            -d read1 \
+            -p "*.R1*.mapstat" \
```

Comment:
Should this be single quotes? Doesn't it have potentially odd interactions here depending on whether `nullglob`/`globstar`/etc are `set` or not? Also _if_ we have to assume filenames here, those need to be documented in the meta (though I'd prefer just not relying on filenames).

Reply:
I'm not sure on the quotes, but you're probably right. The underlying script uses a pattern to find its files, instead of taking them as arguments (which would be much better).